client/clip_client/client.py

```diff
@@ -181,18 +181,17 @@ def _iter_doc(self, content) -> Generator['Document', None, None]:
                 )
 
     def _get_post_payload(self, content, kwargs):
-        return dict(
+        payload = dict(
             on='/',
             inputs=self._iter_doc(content),
             request_size=kwargs.get('batch_size', 8),
             total_docs=len(content) if hasattr(content, '__len__') else None,
-            metadata=_grpc_credential_wrapper(self.credential)
-            if self._scheme == 'grpc'
-            else None,
-            headers=_http_credential_wrapper(self.credential)
-            if self._scheme == 'http'
-            else None,
         )
+        if self._scheme == 'grpc':
+            payload.update(metadata=get_authentication(self._scheme, self.credential))
+        elif self._scheme == 'http':
+            payload.update(headers=get_authentication(self._scheme, self.credential))
+        return payload
```

Comment:
If WebSocket does not accept credential, then let's raise an exception to tell the user.